# Income Gender Gap Analysis

In [1]:
# DEPENDENCIES
import pandas as pd
import numpy as np

In [3]:
# READ IN THE DATA
df = pd.read_csv('inc_occ_gender.csv')

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 558 entries, 0 to 557
Data columns (total 7 columns):
Occupation     558 non-null object
All_workers    558 non-null int64
All_weekly     558 non-null object
M_workers      558 non-null int64
M_weekly       558 non-null object
F_workers      558 non-null int64
F_weekly       558 non-null object
dtypes: int64(3), object(4)
memory usage: 30.6+ KB


In [5]:
df.head() # 

Occupation  All_workers All_weekly  M_workers  \
0                  ALL OCCUPATIONS       109080        809      60746   
1                       MANAGEMENT        12480       1351       7332   
2                 Chief executives         1046       2041        763   
3  General and operations managers          823       1260        621   
4                      Legislators            8         Na          5   

  M_weekly  F_workers F_weekly  
0      895      48334      726  
1     1486       5147     1139  
2     2251        283     1836  
3     1347        202     1002  
4       Na          4       Na

First row doesn't look right. 558 rows with 7 columns, small enough dataset to open and visually inspect
<br> While we see that .info() found no null objects, these are instead just filled with "Na" strings

To remove the Na cells, use a numpy.where() function to replace them with Nulls


In [12]:
# my code here

# first find the object columns
condition = df != 'Na'

df2 = np.where(
    , # condition
    np.nan, # true
    , # false
)

/Users/sampoad/opt/anaconda3/lib/python3.7/site-packages/pandas/core/ops/__init__.py:1115: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  result = method(y)


Then, drop rows that have any Nulls

In [ ]:
# my code here
df3 = df2.dropna(how = 'any', axis = 1')

In [9]:
# there's grouped rows of Occupations
headers = ['MANAGEMENT', 'BUSINESS', 'COMPUTATIONAL', 'ENGINEERING', 'SCIENCE', 'SOCIAL SERVICE', 'LEGAL', 'EDUCATION', 'ARTS', 'HEALTHCARE PROFESSIONAL', 'HEALTHCARE SUPPORT', 'PROTECTIVE SERVICE', 'CULINARY', 'GROUNDSKEEPING', 'SERVICE', 'SALES', 'OFFICE', 'AGRICULTURAL', 'CONSTRUCTION', 'MAINTENANCE', 'PRODUCTION', 'TRANSPORTATION', ]

So I'll need to subset into each of these groups. Show the distribution of gender, and the resulting distribution of income. 